src/error.rs

```diff
@@ -96,3 +106,56 @@ pub(crate) fn check_name(name: &str, tag: &str) -> Result<()> {
         Err(NameError::Invalid(name.to_string(), tag.to_string()).into())
     }
 }
+
+pub(crate) fn check_reset_value(
+    size: &Option<u32>,
```

Comment:
`&Option<u32>`?

Reply:
That's how it's getting stored on RegisterProperties, so I figured I'd just take a reference. I can switch to non-ref if you think it's more ergonomic.

Reply:
I do think a reference to an `Option` of a primitive type  is a weird and unergonomic choice, yes. Not sure how much efforts it would be to change that but I certainly would prefer less ampersands in new code.

Reply:
Sounds good, just pushed with references switched to plain `Option<u32>`s.